src/routes/wallet.ts

```diff
@@ -72,12 +75,14 @@ export async function walletContractHandler(ctx: KoaContext) {
     await Promise.allSettled(
       [...deployedOrOwned].map(async (id: string) =>
         // do not pass any evaluation options, the contract manifests will be fetched for each of these so they properly evaluate
-        (await validateStateWithTimeout({
+        (await validateStateAndOwnership({
           contractTxId: id,
           warp,
           type,
           address,
           logger,
+          // shorten our timeout to 10 seconds for sub-contracts
+          signal: AbortSignal.timeout(SUB_CONTRACT_EVALUATION_TIMEOUT_MS),
```

Comment:
Interesting choice. I'd have thought we'd just use the same top level signal.

Reply:
Good call out - we can sync these